lib/rules/no-cond-assign.js

```diff
@@ -59,7 +59,7 @@ module.exports = {
                 if (isConditionalTestExpression(currentAncestor)) {
                     return currentAncestor.parent;
                 }
-            } while ((currentAncestor = currentAncestor.parent));
+            } while ((currentAncestor = currentAncestor.parent) && currentAncestor.type !== "CallExpression");
```

Comment:
If the assignment is inside a function body, bail out. We don't consider that to be assignment inside a conditional. If the function wasn't inline, we wouldn't even be looking at it.